src/cluster_legacy.c

```diff
@@ -173,10 +173,37 @@ static inline char *clusterLinkGetNodeName(clusterLink *link) {
     return link->node ? link->node->name : "<unknown>";
 }
 
+/* By default, a server doesn't have a human-readable nodename unless explicitly
+ * assigned by CONFIG SET cluster-announce-human-nodename command or config file
+ * edit, so we simply fall back to using the node's IP and port as the nodename.
+ *
+ * WARNING: THIS IS ONLY USED FOR LOGGING PURPOSE.
+ *
+ * Returns either the SDS field or a pointer to a thread-local scratch buffer. */
+char *humanNodename(clusterNode *node) {
```

Comment:
Because of the special limitations, perhaps we should name this function `humanNodenameForLogging` or something like that, to make it clear that it shall not be used for other things.